File: common/src/main/java/com/hartwig/actin/treatment/datamodel/CohortMetadata.java
package com.hartwig.actin.treatment.datamodel;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class CohortMetadata {

    @NotNull
    public abstract String cohortId();

    public abstract boolean evaluable();
    
    public abstract boolean open();

    public abstract boolean slotsAvailable();

    public abstract boolean blacklist();

    @NotNull
    public abstract String description();

}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/trial/CohortMetadata.kt
package com.hartwig.actin.datamodel.trial

typealias Location = String

data class CohortAvailability(val open: Boolean, val slotsAvailable: Boolean)

data class CohortMetadata(
    val cohortId: String,
    val description: String,
    val cohortAvailability: CohortAvailability,
    val availabilityByLocation: Map<Location, CohortAvailability>? = null,
    val evaluable: Boolean,
    val ignore: Boolean,
) {
    val open = cohortAvailability.open
    val slotsAvailable = cohortAvailability.slotsAvailable
}
